app/models/gem_info.rb

```diff
@@ -44,22 +40,39 @@ def self.versions_after(date)
                     v.indexed is false AND
                     v.yanked_at > ?)
               ORDER BY date, number, platform, name", date, date]
+
+    map_gem_version query
+  end
+
+  def self.compact_index_public_versions(date)
+    query = ["SELECT r.name, v.created_at, v.sha256, v.info_checksum, v.number, v.platform
+              FROM rubygems AS r, versions AS v
+              WHERE v.rubygem_id = r.id AND
+                    v.indexed is true AND
+                    v.created_at > ?
+              ORDER BY v.created_at, v.number, v.platform, r.name", date]
+
+    map_gem_version query
+  end
+
+  def self.map_gem_version(query)
     sanitize_sql = ActiveRecord::Base.send(:sanitize_sql_array, query)
     gems = ActiveRecord::Base.connection.execute(sanitize_sql)
 
     gems.map do |gem|
+      version_checksum = Version._sha256_hex(gem['sha256'])
```

Comment:
changing it to hex is unnecessary, the client can handle both hex and base64 https://github.com/bundler/bundler/blob/master/lib/bundler/rubygems_gem_installer.rb#L44